src/services/collections.js

```diff
@@ -1,19 +1,32 @@
-import config from 'config';
-
 import fetch from '../utils/fetch.js';
 
-export const fetchCollections = async () => {
-  const collections = await fetch(config.get('collectionsUrl'));
+export const fetchCollections = async collectionsConfig => {
+  let collections = [];
+
+  for (const item of collectionsConfig) {
+    if (typeof item === 'string') {
+      /* eslint-disable no-await-in-loop */
+      const fetchedCollections = await fetch(item); // Use `await` to ensure sequential fetching of collections and preserve the order in the resulting collections array
+
+      /* eslint-enable no-await-in-loop */
+
+      collections = collections.concat(fetchedCollections.filter(collection => collection.id && collection.name && collection.endpoint));
+    }
 
-  return Object.keys(collections).reduce((result, collectionName) => {
-    if (collections[collectionName].endpoint) {
-      result.push({
-        name: collectionName,
-        id: collections[collectionName].id,
-        endpoint: collections[collectionName].endpoint,
-      });
+    if (typeof item === 'object' && item.id && item.name && item.endpoint) {
+      collections.push(item);
     }
+  }
 
-    return result;
-  }, []);
+  return removeDuplicatesKeepLatest(collections);
 };
+
+export function removeDuplicatesKeepLatest(collections) {
```

Comment:
Use a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead, to keep both unicity and insertion order 🙂

Reply:
Which improvement do you see if I refactor the function by using Map?
```js
export function removeDuplicatesKeepLatest(collections) {
  const uniqueCollections = new Map();

  collections.forEach(collection => {
    uniqueCollections.set(collection.id, collection);
  });

  return Array.from(uniqueCollections.values());
}
```

Reply:
That you can simply get rid of the entire `removeDuplicatesKeepLatest` function. Just insert into a Map in `fetchCollections`.